### PR TITLE
adds functions to identify config formats from filenames

### PR DESCRIPTION
**Pull request by havok2063**
This PR is split from #182 and adds functions for identifying an input file format and application configuration. This functionality is needed by MAST. The idea here is to have some mechanism for identifying an input file format and expected configuration.   It uses the `specutils` `identify_spectrum_format` to attempt to identify a specutils format and if none can be found, uses the dimensionality of the input data to determine a valid configuration.   It's not super general at the moment, focusing on JWST x1d, s2d, s3d and MaNGA cube files.

### CODE REVIEW

**Review by rosteen — COMMENTED** [review]
It surprises me that one of the SDSS files we commonly use for Specviz is unable to be correctly identified by the specutils `identify_spectrum_format` function. I tested your functions with the MaNGA file we commonly use for Cubeviz testing and that worked, but testing on the SDSS file failed (https://data.sdss.org/sas/dr14/sdss/spectro/redux/26/spectra/0751/spec-0751-52251-0160.fits) with `identify_data` returning:
```
([],
 'imviz',
 'Error: Cannot determine format of the file to load.  Please specify a format')
```

I think that at least the error message there needs improvement, since it's unclear where it is directing the user to provide a file format (to the `identify_data` function? when loading the data into the app?).

This PR overall seems good, but I wonder if it should work on more data types before merging, or if MAST is ok with it mainly supporting JWST data for now.

File: jdaviz/core/data_formats.py
Comment (rosteen) [inline jdaviz/core/data_formats.py:115]:
```suggestion
```

Fixes codestyle test error.

**Comment by havok2063** [comment]
@rosteen So there are bugs currently in `specutils` and depending on whether you are using `1.0`, `1.1`, `master` or this PR, https://github.com/astropy/specutils/pull/718, you will see different results.  Sometimes it works with files and not file streams/fits HDULists or vice-versa. Sometimes it works with some gzipped fits files and not unzipped fits, or vice-versa.  This is why I'm pushing to get this PR, https://github.com/astropy/specutils/pull/718 merged as soon as possible.  I believe it fixes all of these issues.  Can you try using that specutils PR and see if you have similar problems?

It is primarily supposed to identify based on whatever formats are registered already in specutils, otherwise it defaults to dimensionality.  I haven't tested this extensively against all cases.  I think currently MAST is concerned with JWST data formats at the moment since this tool is being used for that mission.  However, the next big mission where jdaviz could be applied would be HST data, and I believe that data already has `specutils` data loaders.  You're right that for more general use cases, that data would need to have its own loaders registered in `specutils`.  It may also be true that the `default_mapping` at line 13 would need to get updated.  Do you have any ideas on how to better generalize this?  The file-to-config mapping shouldn't live in `specutils` so we might need to create some custom jdaviz registry that maps a file or set of files to default recommended configurations.

**Comment by rosteen** [comment]
Confirmed that the SDSS file is properly identified with [specutils 718](https://github.com/astropy/specutils/pull/718) installed. I'll do some prodding to see if we can get that moving along and merged, since I'd prefer to merge this after that's in. In the meantime it would be nice to have a couple tests for these functions - I might have time for that tomorrow if you don't, should be pretty quick work.

**Comment by havok2063** [comment]
Glad the specutils PR makes things work for you.  Yeah some tests makes sense.  I'll try to work on adding some.

**Comment by havok2063** [comment]
@rosteen I've added some tests for the functions in `data_formats.py`.  Have a look and let me know what you think.

**Comment by rosteen** [comment]
I just confirmed that the tests pass locally with the master branch of specutils, I'll poke some of those folks about doing a release.

**Review by rosteen — APPROVED** [review]
Confirmed again that this works after updating the setup.cfg file to require dev specutils. Pinged a few other devs for a second review so we can get this merged.

File: jdaviz/core/data_formats.py
Comment (eteq) [inline jdaviz/core/data_formats.py:115]: I'm not clear why a status message is needed here given this is sort of a "library function".  Perhaps instead this:
```suggestion
    if not valid_format:
        raise ValueError('Cannot determine format of the file to load.  Please specify a format')
    elif config not in valid_configs:
        raise ValueError(f"Config {config} not a valid configuration.")
    elif current and config != current:
        raise ValueError('Mismatch between input file format and loaded configuration.')

    return valid_format, config
```
and then the "user" function in MAST instead uses ``try/except`` to catch the error cases?  ala
```
try:
    valid_format, config = identiy_data(...)
    status = 'Success: Valid Format'
except ValueError as e:
    status = 'Error:' + e.args[0]
```

**Review by eteq — CHANGES_REQUESTED** [review]
Overally looks good.  I have a few minor comments, but also one suggestion about not returning a status message.  If there's some motivation I missed I can certainly be talked down from that one though!

**Comment by havok2063** [comment]
Those suggestions look good to me.  I've committed them.

**Review by ibusko — APPROVED** [review]
Tested after all the commits. Works as expected.

**Review by eteq — APPROVED** [review]

**Comment by rosteen** [comment]
Merged!

**Pull request merged**
